### PR TITLE
zpay32: handle segwit prefixes > 2 chars

### PR DESCRIPTION
**Pull request by sangaman**
This is a continuation of #625 with revisions from @halseth to address the bug described in #430.

### CODE REVIEW

File: zpay32/invoice_test.go [hunk 1]
```diff
@@ -90,11 +93,22 @@ var (
...
+	ltcTestNetParams chaincfg.Params
+	ltcMainNetParams chaincfg.Params
 )
 
 func init() {
 	copy(testPaymentHash[:], testPaymentHashSlice[:])
 	copy(testDescriptionHash[:], testDescriptionHashSlice[:])
+
+	// TODO(sangaman): create an interface for chaincfg.params
+	ltcTestNetParams = chaincfg.TestNet3Params
```
Comment (nalinbhardwaj) [inline zpay32/invoice_test.go]: Maybe a comment should be added to clarify what this block is doing? I was mildly confused by it.
Or maybe a better solution is to just pull the [applyLitecoinParams](https://github.com/lightningnetwork/lnd/blob/7abdd30a878efb3673ce4dfe0070570a6c9444c8/chainparams.go#L60) to this test file and use it, but it's ok either way.

Sorry if already discussed somewhere.

Comment (nalinbhardwaj) [reply zpay32/invoice_test.go]: Also, your squashed commit has all the messages merged, maybe modify it to be a single continuous paragraph instead.

Comment (halseth) [reply zpay32/invoice_test.go]: Agree that a comment explaining why/what this block is doing is a good idea 👍

File: zpay32/invoice_test.go [hunk 2]
```diff
@@ -515,10 +529,66 @@ func TestDecodeEncode(t *testing.T) {
...
+					Destination:     testPubKey,
+				}
+			},
+		},
 	}
 
 	for i, test := range tests {
-		invoice, err := Decode(test.encodedInvoice)
+		var decodedInvoice *Invoice
+		var net *chaincfg.Params
```
Comment (halseth) [inline zpay32/invoice_test.go]: You can just set this `= mainnet`, then drop the `else` below.

Comment (sangaman) [reply zpay32/invoice_test.go]: Good point.

**Comment by halseth** [comment]
Looks good! Could you squash into one commit and rebase on latest master? Thanks 👍

**Comment by halseth** [comment]
Actually, could you split them into three distinct commits: `rpcserver`, `zpay32` and `zpay32 test` ?

**Pull request merged**